Remotion/Core/UnitTests/Reflection/TypeDiscovery/AssemblyFinding/AssemblyFinderTest.cs

```diff
@@ -214,11 +211,10 @@ private bool IsAssemblyReferencedBy (Assembly referenced, Assembly origin)
           .Any();
     }
 
-    private static Assembly CompileReferencingAssembly (AssemblyCompilerBuildOutputManager outputManager, Assembly remotionAssembly)
+    private static string CompileReferencingAssembly (AssemblyCompilerBuildOutputManager outputManager, Assembly remotionAssembly)
     {
       var referencingAssemblyRelativePath = outputManager.CompileInSeparateAppDomain("RemotionCoreReferencingAssembly.dll", remotionAssembly.Location);
```

Comment:
When switching the compiler-logic, we need to check if CompileInSeparateAppDomain is still relevant. The basic idea was to prevent the compiled Assembly from getting loaded by accident. It's possible that the CSharpCodeProvider did load the assembly after compilation. If that's not even the case right now, we could also drop this api immediately since here is the only usage.